frontend/app/products/details/page.tsx

```
@@ -0,0 +1,284 @@
+                  <div className="flex gap-2">
+                    {product.colors.map((color) => (
+                      <button
+                        key={color}
+                        onClick={() => setSelectedColor(color)}
+                        className={cn(
+                          "h-8 w-8 rounded-full border-2",
+                          color === "Space Gray" ? "bg-gray-900" : "bg-gray-100",
+                          selectedColor === color ? "ring-2 ring-primary ring-offset-2" : "border-transparent",
+                        )}
+                      >
+                        <span className="sr-only">{color}</span>
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)